Automattic/wp-calypso
# Convos: connect the watermarks

**Summary**
Currently on the conversations stream, users see the latest 3 comments from each post.  It would be much more useful for users to only see unseen comments in the conversations stream.   Keeping track of what has been seen/unseen all falls as part of the umbrella project "High Water Mark".

This pull requests implements high watermark by:
1. if the user is not an Automattician or not on wpcalypso then use old behavior
2. if user does not have a recorded watermark, then use old behavior
3. if we do have a watermark, the only expand the comments after that mark.

It marks the conversations stream as viewed when the component is unmounted.  Is there a more logical time to do this?


To Test
-----
1. boot up convos. should behave as 3 comments per post. Then open a different stream and come back to convos. everything should be collapsed. wait for an orange pill. click orange pill. should do something reasonable.
2. boot up convos as nona8c user, behavior should be back to 3 comments per post.
